modules/app_lb/main.tf

```diff
@@ -62,10 +62,11 @@ resource "azurerm_application_gateway" "default" {
   }
 
   frontend_ip_configuration {
-    name                          = "${local.frontend_ip_configuration_name}-private"
-    subnet_id                     = var.public_subnet.id
```

Comment:
Shouldn't this be private ? So that the ip will be picked up from private subnet